### ISSUE TITLE
Add process definition endpoints' changes

### ISSUE THREAD

**Issue by tmetzke** [camunda/camunda#23741]
## Description

Add the user-facing changes from V1 to V2 to the [API migration guide](https://github.com/camunda/camunda-docs/blob/main/docs/apis-tools/migration-manuals/migrate-to-camunda-api.md).

Suggested procedure:
* Compare the public reference documentation from V1 with the V2 documentation, e.g. the input and output parameters of endpoints, general structure of inputs and outputs, and potential behavioral changes.

```[tasklist]
### Tasks
- [ ] [Search decision definitions](https://docs.camunda.io/docs/next/apis-tools/operate-api/specifications/search-7/) - `POST /v1/decision-definitions/search`
- [ ] [Get decision definition by key](https://docs.camunda.io/docs/next/apis-tools/operate-api/specifications/by-key-6/) - `GET /v1/decision-definitions/:key`
```

**Comment by aleksander-dytko** [comment]
Can we also review and clean up the list of filter parameters in the query endpoint?

![Image](https://github.com/user-attachments/assets/6e2b25b0-e6f7-4599-bdd5-bb27e0ae5e19)

**Comment by tmetzke** [comment]
In what way? 🙂

**Comment by aleksander-dytko** [comment]
@tmetzke I was thinking about `name` and `resourceName` here, but now I wonder - are these two different things?

**Comment by tmetzke** [comment]
Yeah but the descriptions apparently are not really helpful, I would say 😆

**Comment by tmetzke** [comment]
The `name` is the `name` of the process IN the BPMN file and the `resourceName` is the name OF the BPMN file itself.

**Comment by aleksander-dytko** [comment]
Okay, now it makes sense 💃 I thought it came from the param provided in the deploy resource, but I found nothing. 
Can we clarify this somewhere? I guess if it was not clear to me, it might not be obvious to our customers.

**Comment by tmetzke** [comment]
> Can we clarify this somewhere? I guess if it was not clear to me, it might not be obvious to our customers.

I agree. Let's create a separate issue for that, can you do this?

**Comment by aleksander-dytko** [comment]
Issue: https://github.com/camunda/camunda/issues/28551
